engine/src/repository/dataset.rs

```diff
@@ -166,34 +168,44 @@ pub enum DatasetError {
     CreateDatasetFailed,
 }
 
-impl LaerningToolRepository {
-    pub async fn create_dataset(&self, dataset: Dataset) -> Result<Dataset, DatasetError> {
+#[async_trait]
+impl Repository<Dataset, DatasetError> for LaerningToolRepository {
+    async fn create(&self, dataset: Dataset) -> Result<Dataset, RepositoryError<DatasetError>> {
         let mut bindings: HashMap<String, Value> = HashMap::new();
         bindings.insert("data".to_string(), Value::Object(dataset.into()));
+
         let created: Option<Dataset> = self
             .db
             .query("INSERT INTO dataset $data")
             .bind(bindings)
             .await
+            .map_err(|_| RepositoryError::CreationFailed(DatasetError::CreateDatasetFailed))
             .unwrap()
             .take(0)
             .unwrap();
-        created.ok_or(CreateDatasetFailed)
+
+        Ok(created.unwrap())
     }
 
-    pub async fn batch_create_datasets(&self, datasets: Vec<Dataset>) -> Result<(), DatasetError> {
+    async fn create_batch(
```

Comment:
I agree with the naming convention here. Thanks for improving that! I still would like it to name the entity, though.

```suggestion
    async fn create_batch_datasets(
```